pkg/nodeconfig/nodeconfig.go

```diff
@@ -288,6 +304,186 @@ func (nc *nodeConfig) Configure() error {
+	}
+
+	for _, ignFile := range ign.Config.Storage.Files {
+		if _, ok := filesToTranslate[ignFile.Node.Path]; ok {
+			if ignFile.Contents.Source == nil {
+				return errors.Errorf("could not process %s: File is empty", ignFile.Node.Path)
+			}
+			contents, err := dataurl.DecodeString(*ignFile.Contents.Source)
+			if err != nil {
+				return errors.Wrapf(err, "could not decode %s", ignFile.Node.Path)
+			}
+
+			fileName := filepath.Base(ignFile.Node.Path)
+			if err := nc.Windows.EnsureFileContent(contents.Data, fileName, windows.K8sDir); err != nil {
+				return err
+			}
+			nc.log.V(1).Info("created", "file", windows.K8sDir+fileName)
+		}
+	}
+	return nil
+}
+
+// createBootstrapKubeconfig creates a kubeconfig for kubelet to initially communicate with the API server
+func (nc *nodeConfig) createBootstrapKubeconfig() error {
+	nodeBootstrapToken, err := nc.k8sclientset.CoreV1().Secrets(mcoNamespace).Get(context.TODO(), mcoBootstrapSecret,
+		meta.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	kubeconfig, err := generateKubeconfig(nodeBootstrapToken, nodeConfigCache.apiServerEndpoint)
+	if err != nil {
+		return err
+	}
+	kubeconfigData, err := json.Marshal(kubeconfig)
+	if err != nil {
+		return err
+	}
+	kubeconfigFileName := "bootstrap-kubeconfig"
+	if err := nc.Windows.EnsureFileContent(kubeconfigData, kubeconfigFileName, windows.K8sDir); err != nil {
+		return err
+	}
+	nc.log.V(1).Info("created", "file", windows.BootstrapKubeconfig)
+	return nil
+}
+
+// generateKubeconfig creates a kubeconfig spec with the certificate and token data from the given secret
+func generateKubeconfig(secret *core.Secret, apiServerURL string) (*clientcmdv1.Config, error) {
+	// extract ca.crt and token data fields
+	caCert := secret.Data[core.ServiceAccountRootCAKey]
+	if caCert == nil {
+		return nil, errors.New("unable to find CA cert")
+	}
+	token := secret.Data[core.ServiceAccountTokenKey]
+	if token == nil {
+		return nil, errors.New("unable to find token")
+	}
+	kubeconfig := clientcmdv1.Config{
+		Clusters: []clientcmdv1.NamedCluster{{
+			Name: "local",
+			Cluster: clientcmdv1.Cluster{
+				Server:                   apiServerURL,
+				CertificateAuthorityData: caCert,
+			}},
+		},
+		AuthInfos: []clientcmdv1.NamedAuthInfo{{
+			Name: "kubelet",
+			AuthInfo: clientcmdv1.AuthInfo{
+				Token: string(token),
+			},
+		}},
+		Contexts: []clientcmdv1.NamedContext{{
+			Name: "kubelet",
+			Context: clientcmdv1.Context{
+				Cluster:  "local",
+				AuthInfo: "kubelet",
+			},
+		}},
+		CurrentContext: "kubelet",
+	}
+	return &kubeconfig, nil
+}
+
+// createKubeletConf creates config file for kubelet, with Windows specific configuration.
+func (nc *nodeConfig) createKubeletConf(clusterDNS string) error {
+	kubeletConfig := generateKubeletConfiguration(clusterDNS)
+	kubeletConfigData, err := json.Marshal(kubeletConfig)
+	if err != nil {
+		return err
+	}
+	kubletConfigFileName := "kubelet.conf"
```

Comment:
Please define this in the `Windows` package with the rest of the on-node filepaths

Reply:
Oh it looks like it already is there